slsDetectorSoftware/src/Module.cpp

```diff
@@ -1395,14 +1427,25 @@ std::vector<int> Module::getVetoPhoton(const int chipIndex) const {
     }
     int nch = -1;
     client.Receive(&nch, sizeof(nch));
-
-    int adus[nch];
-    memset(adus, 0, sizeof(adus));
-    client.Receive(adus, sizeof(adus));
-    std::vector<int> retvals(adus, adus + nch);
     LOG(logDEBUG1) << "Getting veto photon [" << chipIndex << "]: " << nch
                    << " channels\n";
-    return retvals;
+    int gainIndices[nch];
```

Comment:
While VLAs and memset might be the way to do things in C I would prefer not to use it in C++. Could you maybe read into a vector?

Reply:
done